src/components/ReactFlagsSelect/ReactFlagsSelect.tsx

```diff
@@ -103,6 +107,9 @@ const ReactFlagsSelect: React.FC<Props> = ({
 
     const filteredCountriesOptions = countriesOptions.filter((key) => {
       const label = getLabel(key);
+      if (typeof label === "object") {
```

Comment:
I see we are doing `typeof label === "object"` quite a lot.
Let's expose a utility function in the `/utils` folder, maybe call it `isCustomLabelObject`.
This function can go a further step check if the object has at least the `primary` value.